pyvista/utilities/fileio.py

```diff
@@ -181,21 +181,22 @@ def read(filename, attrs=None, file_format=None):
         return read_meshio(filename, file_format)
 
     # From the extension, decide which reader to use
+    name = os.path.basename(filename)
```

Comment:
We should make sure this is used by other reader routines outside the class init's such as when using `meshio`